clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -149,12 +149,8 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
      * This constrainedAssign optimizes the assignment algorithm when all consumers were subscribed to same set of topics.
      * The method includes the following steps:
      *
-     * 1. Reassign as many previously owned partitions as possible, up to the maxQuota
-     * 2. Fill remaining members up to minQuota
-     * 3. If we ran out of unassigned partitions before filling all consumers, we need to start stealing partitions
-     *    from the over-full consumers at max capacity
-     * 4. Otherwise we may have run out of unfilled consumers before assigning all partitions, in which case we
-     *    should just distribute one partition each to all consumers at min capacity
+     * 1. Reassign as many previously owned partitions as possible, up to the expected number of maxQuota, otherwise, minQuota
```

Comment:
nit: the first line seems should just be `up to the expected number of maxQuota`? Also the second line seems incomplete: we need to explain what condition is `otherwise` on.

Reply:
Nice catch! I put all possibilities in the java doc. Thanks.